packages/react/popover/src/Popover.tsx

```diff
@@ -44,27 +49,60 @@ type PopoverOwnProps = {
 const Popover: React.FC<PopoverOwnProps> = (props) => {
   const { children, open: openProp, defaultOpen, onOpenChange } = props;
   const triggerRef = React.useRef<HTMLButtonElement>(null);
+  const [hasCustomAnchor, setHasCustomAnchor] = React.useState(false);
   const [open = false, setOpen] = useControllableState({
     prop: openProp,
     defaultProp: defaultOpen,
     onChange: onOpenChange,
   });
 
   return (
-    <PopoverProvider
-      contentId={useId()}
-      triggerRef={triggerRef}
-      open={open}
-      onOpenChange={setOpen}
-      onOpenToggle={React.useCallback(() => setOpen((prevOpen) => !prevOpen), [setOpen])}
-    >
-      {children}
-    </PopoverProvider>
+    <PopperPrimitive.Root>
+      <PopoverProvider
+        contentId={useId()}
+        triggerRef={triggerRef}
+        open={open}
+        onOpenChange={setOpen}
+        onOpenToggle={React.useCallback(() => setOpen((prevOpen) => !prevOpen), [setOpen])}
+        hasCustomAnchor={hasCustomAnchor}
+        onCustomAnchorAdd={() => setHasCustomAnchor(true)}
+        onCustomAnchorRemove={() => setHasCustomAnchor(false)}
+      >
+        {children}
+      </PopoverProvider>
+    </PopperPrimitive.Root>
   );
 };
 
 Popover.displayName = POPOVER_NAME;
 
+/* -------------------------------------------------------------------------------------------------
+ * PopoverAnchor
+ * -----------------------------------------------------------------------------------------------*/
+
+const ANCHOR_NAME = 'PopoverAnchor';
+
+type PopoverAnchorOwnProps = Polymorphic.OwnProps<typeof PopperPrimitive.Anchor>;
+type PopoverAnchorPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof PopperPrimitive.Anchor>,
+  PopoverAnchorOwnProps
+>;
+
+const PopoverAnchor = React.forwardRef((props, forwardedRef) => {
+  const context = usePopoverContext(ANCHOR_NAME);
+  const handleCustomAnchorAdd = useCallbackRef(context.onCustomAnchorAdd);
```

Comment:
I think usually we haven't used `useCallbackRef` for these, but rather used `useCallback` when passing into the provider no?

Reply:
These aren't passed into the provider, they're used in the `useEffect`.

Reply:
I know but they come from the provider context and we usually do the `useCallback` there where we pass them in, so then we can pass them directly to dependencies of `useEffect`.

Reply:
Oh yes sorry my bad!